test/DeleteCommandTest.py

```diff
@@ -160,6 +164,47 @@ def test_multi_del5(self):
         self.assertEqual(self.output, "")
         self.assertEqual(self.errors, u("Invalid todo number given: Fo\u00d3B\u0105r.\n"))
 
+    def test_expr_del1(self):
+        command = DeleteCommand(["-e", "@test"], self.todolist, self.out, self.error, None)
+        command.execute()
+
+        result = "Removed: a @test with due:2015-06-03\nRemoved: a @test with +project\n"
+
+        self.assertTrue(self.todolist.is_dirty())
+        self.assertEqual(self.todolist.count(), 2)
+        self.assertEqual(self.output, result)
+        self.assertEqual(self.errors, "")
+
+    def test_expr_del2(self):
+        command = DeleteCommand(["-e", "@test", "due:2015-06-03"], self.todolist, self.out, self.error, None)
+        command.execute()
+
+        self.assertTrue(self.todolist.is_dirty())
+        self.assertEqual(self.output, "Removed: a @test with due:2015-06-03\n")
+        self.assertEqual(self.errors, "")
+
+    def test_expr_del3(self):
+        command = DeleteCommand(["-e", "@test", "due:2015-06-03", "+project"], self.todolist, self.out, self.error, None)
+        command.execute()
+
+        self.assertFalse(self.todolist.is_dirty())
+
+    def test_expr_del4(self):
+        """ Remove only relevant todo items. """
+        command = DeleteCommand(["-e", ""], self.todolist, self.out, self.error, None)
+        command.execute()
+
+        self.assertTrue(self.todolist.is_dirty())
+        self.assertEqual(self.todolist.count(), 1)
```

Comment:
Same: checking just the count is not enough, also check the content of what's left.